public/app/features/alerting/unified/rule-list/FilterView.tsx

```diff
@@ -0,0 +1,127 @@
+import { take, tap, withAbort } from 'ix/asynciterable/operators';
+import { useEffect, useRef, useState, useTransition } from 'react';
+import Skeleton from 'react-loading-skeleton';
+
+import { Card, EmptyState, Stack, Text } from '@grafana/ui';
+import { Trans } from 'app/core/internationalization';
+
+import { isLoading, useAsync } from '../hooks/useAsync';
+import { RulesFilter } from '../search/rulesSearchParser';
+
+import { AlertRuleLoader } from './AlertRuleLoader';
+import LoadMoreHelper from './LoadMoreHelper';
+import { ListItem } from './components/ListItem';
+import { ActionsLoader } from './components/RuleActionsButtons.V2';
+import { RuleListIcon } from './components/RuleListIcon';
+import { RuleWithOrigin, useFilteredRulesIteratorProvider } from './hooks/useFilteredRulesIterator';
+
+interface FilterViewProps {
+  filterState: RulesFilter;
+}
+
+const FRONTENT_PAGE_SIZE = 100;
+const API_PAGE_SIZE = 2000;
+
+export function FilterView({ filterState }: FilterViewProps) {
+  // ⚠️ We use a key to force the component to unmount and remount when the filter state changes
```

Comment:
Why is do we want to unmount and mount? is not enough the filterState prop in `FilterViewResults` to render the changes?

Reply:
I will improve the comment
`filterState` is a quite complex object constructed from URL query params, so we might get a new reference to it even though query parameters haven't actually changed
On the other hand, we use `key` to reset the component, mainly because we want to reset generators when filters changed